tests/cross_functional/ui/fusion_access/test_virtualmachine_lifecycle.py

```
@@ -45,6 +60,93 @@ def setup_ui(self, setup_ui_class_factory):
+            encoding="utf-8",
+            timeout=120,
+        )
+
+        # Wait for the "Press Ctrl" banner confirming the console is connected,
+        # then send Enter to wake the terminal and trigger the login prompt.
+        child.expect(r"Press Ctrl", timeout=60)
+        child.sendline("")
+
+        # The VM may still be booting — wait up to 5 minutes for the login prompt.
+        child.expect(r"login:", timeout=300)
+        logger.info(f"Login prompt detected — sending username: '{vm_username}'")
```

Ruff (0.16.0)

[warning] Logging statement uses f-string

(G004)